opensplice_cmake_module/env_hook/opensplice.sh.in

```diff
@@ -1,16 +1,29 @@
 export OSPL_HOME=@OpenSplice_HOME@
 
+_DEFAULT_OSPL_URI="file://@CMAKE_INSTALL_PREFIX@/share/opensplice_cmake_module/config/ros_ospl.xml"
+
 if [ -f "$OSPL_HOME/release.com" ]; then
+  _ORIGINAL_OSPL_URI="$OSPL_URI"
   . "$OSPL_HOME/release.com" > /dev/null
-else
-  export OSPL_TMPL_PATH=$OSPL_HOME/etc/opensplice/idlpp
-  _DEFAULT_OSPL_URI=file://$OSPL_HOME/etc/opensplice/config/ospl.xml
-  if [ -z "$OSPL_URI" ]; then
-    export OSPL_URI=$_DEFAULT_OSPL_URI
-  elif [ "$OSPL_URI" != "$_DEFAULT_OSPL_URI" ]; then
-    echo "Warning: OSPL_URI was already set to [[$OSPL_URI]]. This will not override it to the default [[$_DEFAULT_OSPL_URI]]. Please make sure this is the config that you want." 1>&2
+  if [ -n "$_ORIGINAL_OSPL_URI" ]; then
+    export OSPL_URI="$_ORIGINAL_OSPL_URI"
+  else
+    export OSPL_URI="$_DEFAULT_OSPL_URI"
   fi
-  unset _DEFAULT_OSPL_URI
+  unset _ORIGINAL_OSPL_URI
+fi
+
+if [ -z "$OSPL_URI" ]; then
+  export OSPL_URI="$_DEFAULT_OSPL_URI"
+elif [ "$OSPL_URI" != "$_DEFAULT_OSPL_URI" ]; then
+  echo "Warning: OSPL_URI was already set to [[$OSPL_URI]]. This will not override it to the default [[$_DEFAULT_OSPL_URI]]. Please make sure this is the config that you want." 1>&2
+fi
+
+unset _DEFAULT_OSPL_URI
+
+# If release.com was not sourced...
+if [ -z "$OSPL_TMPL_PATH" ]; then
+  export OSPL_TMPL_PATH=$OSPL_HOME/etc/opensplice/idlpp
```

Comment:
Line 26 was before done in line 6 as part of an else block. I think it should stay the same - in an else block of line 5 which would be inserted at line 14.